File: Java/Ejercicio 1/GestionVocabulario.java
import java.util.ArrayList;
import java.util.List;

class Palabra {
    String palabra;
    String significado;
    int aciertos;
    int errores;

    Palabra(String palabra, String significado) {
        this.palabra = palabra.toLowerCase();
        this.significado = significado.toLowerCase();
        this.aciertos = 0;
        this.errores = 0;
    }
}

public class GestionVocabulario {
    public static List<Palabra> vocabulario = new ArrayList<>();

    public static void agregarPalabra(String palabra, String significado) {
        vocabulario.add(new Palabra(palabra, significado));
        System.out.println("Palabra agregada: " + palabra);
    }

    public static Palabra buscarPalabra(String palabra) {
        for (Palabra p : vocabulario) {
            if (p.palabra.equalsIgnoreCase(palabra)) {
                System.out.println("Palabra buscada: " + p.palabra + " -> " + p.significado);
                return p;
            }
        }
        System.out.println("Palabra no encontrada");
        return null;
    }

    public static void editarPalabra(String palabra, String nueva, String significado) {
        Palabra p = buscarPalabra(palabra);
        if (p != null) {
            p.palabra = nueva.toLowerCase();
            p.palabra = significado.toLowerCase();
            System.out.println("Palabra editada: " + nueva);
        }
    }

    public static void eliminarPalabra(String palabra) {
        Palabra p = buscarPalabra(palabra);
        if (p != null) {
            vocabulario.remove(p);
            System.out.println("Palabra eliminada: " + palabra);
        }
    }
}

File: Typescript/Ejercicio 1/gestionVocabulario.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.vocabulario = void 0;
exports.agregarPalabra = agregarPalabra;
exports.buscarPalabra = buscarPalabra;
exports.editarPalabra = editarPalabra;
exports.eliminarPalabra = eliminarPalabra;
exports.vocabulario = [];
function agregarPalabra(p, s) {
    exports.vocabulario.push({
        palabra: p.toLowerCase(),
        significado: s.toLowerCase(),
        aciertos: 0,
        errores: 0
    });
    console.log("Palabra \"".concat(p, "\" agregada."));
}
function buscarPalabra(p) {
    var item = exports.vocabulario.find(function (x) { return x.palabra === p.toLowerCase(); });
    if (item) {
        console.log("".concat(item.palabra, " -> ").concat(item.significado));
        return item;
    }
    console.log("No encontrada.");
    return null;
}
function editarPalabra(p, np, ns) {
    var item = buscarPalabra(p);
    if (item) {
        item.palabra = np.toLowerCase();
        item.significado = ns.toLowerCase();
        console.log("Editada a ".concat(np));
    }
}
function eliminarPalabra(p) {
    var index = exports.vocabulario.findIndex(function (x) { return x.palabra === p.toLowerCase(); });
    if (index >= 0) {
        exports.vocabulario.splice(index, 1);
        console.log("Eliminada ".concat(p));
    }
    else {
        console.log("No encontrada.");
    }
}
